cmake/api_macros.cmake

```diff
@@ -191,6 +191,11 @@ macro(OPENDDS_TARGET_SOURCES target)
     list(APPEND _tao_options "-SS")
   endif()
 
+  if(NOT ${OPENDDS_COMPILER} STREQUAL "cl" AND ${OPENDDS_HOST} STREQUAL "win32")
+    list(APPEND _opendds_options "-Yp,${CMAKE_CXX_COMPILER}")
+    list(APPEND _tao_options "-Yp,${CMAKE_CXX_COMPILER}")
```

Comment:
To allow for overriding it should check the options before appending, like above for the `-Wb,export_*` options and the `--default-nested` options.